src/coreComponents/finiteVolume/SurfaceElementStencil.hpp

```diff
@@ -116,12 +116,30 @@ class SurfaceElementStencilWrapper : public StencilWrapperBase< SurfaceElementSt
   /**
    * @brief Compute weights and derivatives w.r.t to one variable.
    * @param[in] iconn connection index
+   * @param[in] ip phase index
    * @param[in] coefficient view accessor to the coefficient used to compute the weights
    * @param[in] dCoeff_dVar view accessor to the derivative of the coefficient w.r.t to the variable
    * @param[out] weight view weights
    * @param[out] dWeight_dVar derivative of the weights w.r.t to the variable
    */
   GEOS_HOST_DEVICE
+  void computeWeights( localIndex const iconn,
+                       localIndex const ip,
+                       CoefficientAccessor< arrayView4d< real64 const > > const & coefficient,
+                       CoefficientAccessor< arrayView4d< real64 const > > const & dCoeff_dVar,
+                       real64 ( & weight )[maxNumConnections][2],
+                       real64 ( & dWeight_dVar )[maxNumConnections][2] ) const
+  { GEOS_UNUSED_VAR( iconn, ip, coefficient, dCoeff_dVar, weight, dWeight_dVar ); };
```

Comment:
If necessary for diffusion this function should also be implemented for the other stencil types. All the assembly kernels should work with any stencil type so that all physics can be used in the faults/fractures. I don't think any special treatment is needed.

Reply:
@CusiniM I did a first pass at refactoring stencils to fit the proposed formalism. I did it quite blindly and second pass should come up with some more efficient refactoring. (There is quite some overloads in `SurfaceElementStencil`, `EmbeddedSurfaceToCellStencil` and `FaceElementToCellStencil`). Please review and let me know your feedback. Two point prior though:

1. I did not include 
```
GEOS_HOST_DEVICE
inline void
SurfaceElementStencilWrapper::
  computeWeights( localIndex iconn,
                  CoefficientAccessor< arrayView3d< real64 const > > const & coefficient1,
                  CoefficientAccessor< arrayView3d< real64 const > > const & coefficient1Multiplier,
                  CoefficientAccessor< arrayView1d< real64 const > > const & coefficient2,
                  R1Tensor const & unitGravityVector,
                  real64 ( & weight1 )[maxNumPointsInFlux],
                  real64 ( & weight2 )[maxNumPointsInFlux],
                  real64 ( & geometricWeight )[maxNumPointsInFlux] ) const
```
or

```
GEOS_HOST_DEVICE
inline void
SurfaceElementStencilWrapper::
  computeWeights( localIndex iconn,
                  CoefficientAccessor< arrayView3d< real64 const > > const & coefficient,
                  CoefficientAccessor< arrayView3d< real64 const > > const & coefficientMultiplier,
                  R1Tensor const & gravityVector,
                  real64 (& weight)[maxNumConnections][2] ) const
```
as I did not find usage for second and only find usage of first in propant solver. These seems special cases but could be refactor _if needed_.

2. I might have overdone in adding an overload for 
```
GEOS_HOST_DEVICE
inline void
FaceElementToCellStencilWrapper::computeWeights( const localIndex iconn,
                                                 const localIndex ip,
                                                 const CoefficientAccessor< arrayView4d< const real64 > > & coefficient,
                                                 const CoefficientAccessor< arrayView4d< const real64 > > & dCoeff_dVar1,
                                                 const CoefficientAccessor< arrayView4d< const real64 > > & dCoeff_dVar2,
                                                 real64 (& weight)[1][2],
                                                 real64 (& dWeight_dVar1)[1][2],
                                                 real64 (& dWeight_dVar2)[1][2] ) const
```
This might never be used with phase-indexed coefficient, so let me know and I'll revert it

Reply:
If I understand correctly what's happening, I think all stencils need to be able to deal with a coefficient that depends on the phase index. However, I think they all deal with it in the same way so most things can be implemented in the base class and what needs to be specialized is the method `computeWeightsBase`